katcp/resource_client.py

```diff
@@ -1380,10 +1380,12 @@ def call_in_ioloop(self, fn, args, kwargs, timeout=None):
             # Should never get here since the tornado future should raise
             assert False, 'Tornado Future should have raised'
 
-    def decorate_callable(self, callable_, timeout=None):
+    def decorate_callable(self, callable_):
         """Decorate a callable to use call_in_ioloop"""
         @wraps(callable_)
         def decorated(*args, **kwargs):
+            # Extract timeout from request itself or use default for ioloop wrapper
+            timeout = kwargs.get('timeout')
```

Comment:
`get` will return empty string as default which might not be interpreted correctly in the callable ... would suggest explicit `None` as default when timeout argument not present.